spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SparkBinPackFileRewriteRunner.java

```diff
@@ -56,16 +58,18 @@ protected void doRewrite(String groupId, List<FileScanTask> group) {
         .write()
         .format("iceberg")
         .option(SparkWriteOptions.REWRITTEN_FILE_SCAN_TASK_SET_ID, groupId)
-        .option(SparkWriteOptions.TARGET_FILE_SIZE_BYTES, writeMaxFileSize())
-        .option(SparkWriteOptions.DISTRIBUTION_MODE, distributionMode(group).modeName())
-        .option(SparkWriteOptions.OUTPUT_SPEC_ID, outputSpecId())
+        .option(SparkWriteOptions.TARGET_FILE_SIZE_BYTES, group.maxOutputFileSize())
+        .option(
+            SparkWriteOptions.DISTRIBUTION_MODE,
+            distributionMode(group.fileScanTasks(), spec(group.outputSpecId())).modeName())
+        .option(SparkWriteOptions.OUTPUT_SPEC_ID, group.outputSpecId())
         .mode("append")
         .save(groupId);
   }
 
   // invoke a shuffle if the original spec does not match the output spec
-  private DistributionMode distributionMode(List<FileScanTask> group) {
-    boolean requiresRepartition = !group.get(0).spec().equals(outputSpec());
+  private DistributionMode distributionMode(List<FileScanTask> group, PartitionSpec outputSpec) {
```

Comment:
rather than adding more parameters and extracting more fields from "group" why not just pass group all the way through? Would simplify line 64 as well

Reply:
Good idea.
Done